node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala

```diff
@@ -9,34 +9,64 @@ import com.wavesplatform.lang.v1.compiler.Terms
 import com.wavesplatform.lang.v1.estimator.v3.ScriptEstimatorV3
 import com.wavesplatform.transaction.smart.script.ScriptCompiler
 
-class PseudoTxSenderPubKeySuite extends BaseTransactionSuite {
+class PseudoTransactionSuite extends BaseTransactionSuite {
 
-  private def firstDApp     = firstKeyPair
-  private def secondDApp    = secondKeyPair
-  private def caller        = thirdKeyPair
-  private var firstAssetId  = ""
+  private def firstDApp = firstKeyPair
+
+  private def secondDApp = secondKeyPair
+
+  private def caller = thirdKeyPair
+
+  private var firstAssetId = ""
   private var secondAssetId = ""
+
   protected override def beforeAll(): Unit = {
     super.beforeAll()
+    firstAssetId = sender.issue(firstDApp, fee = issueFee, script = Some(scriptBase64), waitForTx = true).id
+    secondAssetId = sender.issue(secondDApp, fee = issueFee, script = Some(scriptBase64), waitForTx = true).id
+
     val smartAssetScript = ScriptCompiler(
       s"""
          |{-# STDLIB_VERSION 4 #-}
          |{-# CONTENT_TYPE EXPRESSION #-}
          |{-# SCRIPT_TYPE ASSET #-}
          |
          |  match tx {
-         |    case t: TransferTransaction => t.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |    case r: ReissueTransaction => r.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |    case b: BurnTransaction => b.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |
+         |    case t: TransferTransaction => t.senderPublicKey.toBase58String() == "${firstDApp.publicKey.toString}"
+         |     && t.assetId.value().toBase58String() == "$firstAssetId"
+         |     && toBase64String(t.attachment) == ""
+         |     && t.bodyBytes.size() == 0
+         |     && t.fee == 0
+         |     && t.id.size() != 0
```

Comment:
can we calculate it and check exact value?

Reply:
done